### ISSUE TITLE
Exception: Failed to look up entry car_0 in catalog VehicleCatalog

### ISSUE THREAD

**Issue by johschmitz** [esmini/esmini#600]
I am getting this weird error which I cannot explain. I tried to compare with another working scenario and did not find a relevant difference.

Here the two scenarios:

[example_exception.zip](https://github.com/user-attachments/files/16482585/example_exception.zip)

Working example from the other issue:
https://github.com/user-attachments/files/16147987/example.zip

**Comment by eknabevcc** [comment]
I agree. Seems to be no significant difference. And I can't reproduce error. 

- Do you run the two scenarios in the same way?
- What is the current folder at execution, in relation to "bdsc" scenario folder.
- What esmini version?

The relative paths, in scenario file and catalog looks correct. Could it be that esmini finds and use a copy of the scenario located at a different location than intended, hence relative paths failing?

**Comment by johschmitz** [comment]
Ah that is a good hint, maybe something like that was the case. I remember that I had a PR long ago before you switched to C++17 that tried to avoid such issues. I need to revisit that maybe.

**Comment by johschmitz** [comment]
I found it: #403

**Comment by eknabevcc** [comment]
That's very true! And there is a branch that we worked on. 
https://github.com/esmini/esmini/tree/feature/pr_403_rb
It nearly made it to be merged with master. I think there was a tiny issue with std::filesystem lacking some specific function in one of the compiler versions. Also, this branch was the main motivation for bumping to C++ v17. 

I still would like to do this at some point. But it would be some work to rebase it. And do the last details (which usually turns out to be slightly more than expected).

I remember that when I wrote the [User guide updates](https://github.com/esmini/esmini/blob/feature/pr_403_rb/docs/user_guide.adoc#811-resources-not-found) I realized the complexity. On one hand we're trying to be flexible, on the other hand propose a default structure. 

Maybe there should be a re-take on handling of input and resource files.

**Comment by johschmitz** [comment]
I checked and can now confirm that this issue was caused by directories above the current working directory. So this would be fixed by the changes we discussed around #403. Let's see if I remember this problem in 6 month or so when I accidentally run into it again :D Maybe at that point the compiler issue blocker is history and we can pick this up again.

**Comment by eknabevcc** [comment]
Thanks for checking and confirming the root cause!
Yes, I'd really like to improve the file path handling according to ideas in the branch. Either the c++ lib blocker becomes history, or we can find a way around it. But also we should make sure backward compatibility. I hope we can find the time.
